### PR TITLE
MAINT: Update sphinx-external-toc version pinning to be a range

### PR DESCRIPTION
**Pull request by mmcky**
This looks to address https://github.com/executablebooks/jupyter-book/pull/1321 and #59 (a circular dependency)

This PR also updates fixtures which I presume have changed due to an update in `sphinx` 
@AakashGfude can we review these file regressions?

- [x] check file regression updates from latest test sets

### CODE REVIEW

**Comment by chrisjsewell** [comment]
> This PR also updates fixtures which I presume have changed due to an update in sphinx

More likely form docutils 0.17, I would double check that the output is still the same, in case we need to pin to <0.17

File: tests/test_sphinx/test_build_no_ext.resolved.xml [hunk 1]
```diff
@@ -26,17 +26,21 @@
                         <paragraph>
                             <inline ids="id6">
                                 [
-                                <citation_reference docname="part1/chap2" refname="id5">
```
Comment (mmcky) [inline tests/test_sphinx/test_build_no_ext.resolved.xml:29]: @AakashGfude I am most concerned by this change. It seems to be not resolving a `citation_reference`

**Comment by mmcky** [comment]
> More likely form docutils 0.17, I would double check that the output is still the same, in case we need to pin to <0.17

Hmm -- using `tox -e py38-sphinx3` to run the tests so `docutils==0.16` in the listings.

**Comment by mmcky** [comment]
I am having bad luck with `testing infrastructure` -- so `pytest` runs (as is in the `ci.yml`) but `tox -e py38-sphinx3` doesn't run.

**Comment by mmcky** [comment]
OK these changes are coming from `sphinxcontrib-bibtex`

https://sphinxcontrib-bibtex.readthedocs.io/en/latest/changes.html#may-2021

**Comment by mmcky** [comment]
@chrisjsewell just updated this to include sphinxcontrib-bibtex as a range

```
sphinxcontrib-bibtex>=2.2.1,<2.3.0
```

**Comment by choldgraf** [comment]
feel free to ping me once this is merged and I will release a new JB

**Comment by mmcky** [comment]
@chrisjsewell I will merge this today and then setup a PR to update the name of this package.

**Pull request merged**